### ISSUE TITLE
Redundant 

### ISSUE THREAD

**Issue by Rekkonnect** [dotnet/roslyn#72101]
**Version Used**:
VS 2022 Version 17.9.0
Roslyn 4.9.0 - 989117396f26e

**Steps to Reproduce**:

1. Define a constant field in a class that contains a circular reference with itself (this makes its constant value uncomputable)
2. Use that constant field as an attribute argument, matching the type of the attribute parameter
3. Observe the errors on the attribute argument

[Repro on SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAFxANwKYCcoDMCeAfAAQCYBGAWAChKBtAZQHsxUAKANQgBsBXVASgF0aDJm048SAygQDMAAgIkAbPKKzalAN6VZO+XIDG9GAGcEs2GfbdUsgLyyrPANzbdM2YZNmLDsahJ2vtayANSyJC4UAL6UUnLGqJyoACYqaoyoAIIICJjAXAgsPsh+vLIgstm5UPmFTkA=)

**Expected Behavior**:

The attribute argument does not cause any errors.

**Actual Behavior**:

Using the circular constant as an attribute argument yields an error about not being constant.

**Notes**:

This only applies to attribute arguments that require constant values. Using a constant field containing a circular reference in the definition of another constant field, unrelated to the circular reference graph, does **not** report any errors, which is demonstrated in the same repro linked above. This is more preferable to correctly isolate the uncomputability of the field in the circular reference, and not the attribute argument's value.

**Comment by CyrusNajmabadi** [comment]
This seems like a very niche scenario.  Basically an error has to happen in one place, for another error to arise.  That first error itself would be pretty uncommon.

**Comment by Rekkonnect** [comment]
Indeed, it's barely any significant. I'd be happy to take up the fix if that's okay.

**Comment by CyrusNajmabadi** [comment]
@jcouv @jaredpar  Any concerns about @Rekkonnect tweaking the compiler to suppress cascaded diagnostics in this niche case?

**Comment by jaredpar** [comment]
> error CS0110: The evaluation of the constant value for 'S.Value' involves a circular definition

Even if the situation is niche changing circularity detection in the compiler is likely to be complex and undergo a high degree of scrutiny by the team. 

> ... tweaking the compiler to suppress cascaded diagnostics in this niche case?

Said another way: circularity is a complex area of the compiler that we fairly regularly regress. It's hard to justify changing that for a niche scenario.
